engine/resources/mount_test.go

```diff
@@ -0,0 +1,273 @@
+		[]byte(fstabMock1),
+		&fstab.Mount{
+			Spec:    "UUID=ef5726f2-615c-4350-b0ab-f106e5fc90ad",
+			File:    "/",
+			VfsType: "ext4",
+			MntOps:  map[string]string{"defaults": ""},
+			Freq:    1,
+			PassNo:  1,
+		},
+		true,
+	},
+	{
+		[]byte(fstabMock1),
+		&fstab.Mount{
+			Spec:    "/dev/mapper/root",
+			File:    "/home",
+			VfsType: "ext4",
+			MntOps:  map[string]string{"defaults": ""},
+			Freq:    1,
+			PassNo:  1,
+		},
+		false,
+	},
+}
+
+func TestFstabEntryExists(t *testing.T) {
+	file, err := ioutil.TempFile("", "fstab")
+	if err != nil {
+		t.Error(err)
+	}
+	defer os.Remove(file.Name())
+
+	for _, test := range fstabEntryExistsTests {
+		if err := ioutil.WriteFile(file.Name(), test.fstabMock, 0644); err != nil {
+			t.Error(err)
+		}
+		result, err := fstabEntryExists(file.Name(), test.in)
+		if err != nil {
+			t.Error(err)
+		}
+		if result != test.out {
+			t.Errorf("fstabEntryExists test wanted: %t, got: %t", test.out, result)
+		}
+	}
+}
+
+var fstabWriteTests = []struct {
+	in fstab.Mounts
+}{
+	{
+		fstab.Mounts{
+			&fstab.Mount{
+				Spec:    "/dev/cdrom",
+				File:    "/mnt/cdrom",
+				VfsType: "iso9660",
+				MntOps:  map[string]string{"defaults": ""},
+			},
+		},
+	},
+}
+
+func TestFstabWrite(t *testing.T) {
+	file, err := ioutil.TempFile("", "fstab")
+	if err != nil {
+		t.Error(err)
+	}
+	defer os.Remove(file.Name())
+
+	for _, test := range fstabWriteTests {
+		if err := fstabWrite(file.Name(), test.in); err != nil {
+			t.Error(err)
+		}
+		for _, mount := range test.in {
+			exists, err := fstabEntryExists(file.Name(), mount)
+			if err != nil {
+				t.Error(err)
+			}
+			if !exists {
+				t.Errorf("failed to write\n%+v\n to fstab", mount)
+			}
+		}
+	}
+}
+
+var fstabEntryAddTests = []struct {
+	fstabMock []byte
+	in        *fstab.Mount
+}{
+	{
+		[]byte(fstabMock1),
+		&fstab.Mount{
+			Spec:    "/dev/sdb1",
+			File:    "/mnt/foo",
+			VfsType: "ext2",
+			MntOps:  map[string]string{"ro": "", "blocksize": "2048"},
+		},
+	},
+}
+
+func TestFstabEntryAdd(t *testing.T) {
+	file, err := ioutil.TempFile("", "fstab")
+	if err != nil {
+		t.Error(err)
+	}
+	defer os.Remove(file.Name())
+
+	for _, test := range fstabEntryAddTests {
+		if err := ioutil.WriteFile(file.Name(), test.fstabMock, 0644); err != nil {
+			t.Error(err)
+		}
+		err := fstabEntryAdd(file.Name(), test.in)
+		if err != nil {
+			t.Error(err)
+		}
+		exists, err := fstabEntryExists(file.Name(), test.in)
+		if !exists {
+			t.Errorf("fstab failed to add entry: %+v to fstab", test.in)
+		}
+	}
+}
+
+var fstabEntryRemoveTests = []struct {
+	fstabMock []byte
+	in        *fstab.Mount
+}{
+	{
+		[]byte(fstabMock1),
+		&fstab.Mount{
+			Spec:    "UUID=ef5726f2-615c-4350-b0ab-f106e5fc90ad",
+			File:    "/",
+			VfsType: "ext4",
+			MntOps:  map[string]string{"defaults": ""},
+			Freq:    1,
+			PassNo:  1,
+		},
+	},
+}
+
+func TestFstabEntryRemove(t *testing.T) {
+	file, err := ioutil.TempFile("", "fstab")
+	if err != nil {
+		t.Error(err)
```

Comment:
what would happen if this errored? ... the code would continue running which would be bad. So you need to `return` right after.
If possible replace with Errorf.